visidata/basesheet.py

```diff
@@ -91,7 +91,7 @@ def options(self):
 
     def __init__(self, *names, **kwargs):
         self._name = None   # initial cache value necessary for self.options
-        self.names = names
+        self.names = list(names)
```

Comment:
I was looking at this report and found the same bugs.  I do not know the code very well, and was wondering does the getter/setter from visidata/sheets line 949  work?  Will changing the name from the Index sheet change the name of the file that is saved?  It updates names[] but it doesn't update name.  I don't know how the two fields names and name are supposed to interact. It seems like one is calculated from the other looking this line and the next.

https://github.com/saulpw/visidata/blob/fb0a6ad18247586b9c0a6b38e708abfebd14dc44/visidata/sheets.py#L949

Reply:
You are absolutely right! It changes the sheet name through the `SheetsSheet`, but not for html tables, at least.

Reply:
Thank you! I think I got a much better fix now.

Reply:
Glad I could help!  Yes, I like this fix better, and I learned something.